src/pages/MatchCreation/MatchCreation.tsx

```diff
@@ -54,6 +58,11 @@ export const Match = () => {
     }
 
     mutate(match)
+
+    if (isSuccess) {
```

Comment:
I don't think this will work. `mutate(match)` is an asynchronous call, so it will just start the promise and immediately jump into this if statement, and `isSuccess` will not have been updated yet.

Later, when the request was successful, the query will change the state of `isSuccess` to true and the component will re-render.

So my suggestion here would be to wrap `isSuccess` with a `useEffect` and then navigate based on if the query was sucessfull.